compiler/src/jdk.graal.compiler/src/jdk/graal/compiler/core/common/type/Stamp.java

```diff
@@ -272,7 +285,7 @@ public boolean neverDistinct(Stamp other) {
         Constant constant = this.asConstant();
         if (constant != null) {
             Constant otherConstant = other.asConstant();
-            return otherConstant != null && constant.equals(otherConstant);
```

Comment:
why drop the null check here?

Reply:
Think it was automatically refactored by the IDE, I will insert the null check again